src/SFML/Window/Win32/JoystickImpl.hpp

```diff
@@ -117,111 +102,33 @@ class JoystickImpl
-    ///
-    /// \return `true` if the joystick is connected, `false` otherwise
-    ///
-    ////////////////////////////////////////////////////////////
-    static bool isConnectedDInput(unsigned int index);
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Update the connection status of all joysticks (DInput)
-    ///
-    ////////////////////////////////////////////////////////////
-    static void updateConnectionsDInput();
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Open the joystick (DInput)
-    ///
-    /// \param index Index assigned to the joystick
-    ///
-    /// \return `true` on success, `false` on failure
-    ///
-    ////////////////////////////////////////////////////////////
-    [[nodiscard]] bool openDInput(unsigned int index);
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Close the joystick (DInput)
-    ///
-    ////////////////////////////////////////////////////////////
-    void closeDInput();
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Get the joystick capabilities (DInput)
-    ///
-    /// \return Joystick capabilities
-    ///
-    ////////////////////////////////////////////////////////////
-    [[nodiscard]] JoystickCaps getCapabilitiesDInput() const;
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Update the joystick and get its new state (DInput, Buffered)
-    ///
-    /// \return Joystick state
-    ///
-    ////////////////////////////////////////////////////////////
-    [[nodiscard]] JoystickState updateDInputBuffered();
-
-    ////////////////////////////////////////////////////////////
-    /// \brief Update the joystick and get its new state (DInput, Polled)
-    ///
-    /// \return Joystick state
-    ///
-    ////////////////////////////////////////////////////////////
-    [[nodiscard]] JoystickState updateDInputPolled();
+    [[nodiscard]] JoystickState update() const;
 
 private:
-    ////////////////////////////////////////////////////////////
-    /// \brief Device enumeration callback function passed to EnumDevices in updateConnections
-    ///
-    /// \param deviceInstance Device object instance
-    /// \param userData       User data (unused)
-    ///
-    /// \return DIENUM_CONTINUE to continue enumerating devices or DIENUM_STOP to stop
-    ///
-    ////////////////////////////////////////////////////////////
-    static BOOL CALLBACK deviceEnumerationCallback(const DIDEVICEINSTANCE* deviceInstance, void* userData);
+    /// \brief informs system of device added
+    static void dispatchDeviceConnected(HANDLE deviceHandle);
+    /// \brief informs system of device removed
+    static void dispatchDeviceRemoved(HANDLE deviceHandle);
+    /// \brief Performs XInput Operations
+    static void dispatchXInput();
+    /// \brief Performs RawInput Operations
```

Comment:
Eventually you'll need to add the same doxygen headers that all other functions have which means tweaking the formatting and adding docs for each parameter.